File: src/main/java/itmo/is/project/model/user/Account.java
package itmo.is.project.model.user;

import jakarta.persistence.*;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.*;

@Entity
@Table(name = "account")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Account {

    @Id
    private Integer userId;

    @MapsId
    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private User user;

    @NotNull
    @Min(0)
    @Column(name = "balance", nullable = false)
    private Integer balance = 0;

    public void deposit(Integer amount) {
        balance += amount;
    }

    public void withdraw(Integer amount) {
        balance -= amount;
    }
}

File: src/main/kotlin/itmo/isproject/model/user/Account.kt
package itmo.isproject.model.user

import itmo.isproject.model.IntIdEntity
import jakarta.persistence.*
import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotNull

@Entity
@Table(name = "account")
class Account(

    @MapsId
    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    var user: User? = null,

    @NotNull
    @Min(0)
    @Column(name = "balance", nullable = false)
    var balance: Int = 0

) : IntIdEntity() {

    fun deposit(amount: Int?) {
        balance += amount ?: 0
    }

    fun withdraw(amount: Int?) {
        balance -= amount ?: 0
    }
}
